### PR TITLE
Add support for impersonation tokens

### PR DESCRIPTION
**Pull request by maxenced**
As there is no deletion in gitlab api, we do not update and use delete to revoke the token.

### CODE REVIEW

File: gitlab/resource_gitlab_project_hook.go [hunk 1]
```diff
@@ -118,6 +118,7 @@ func resourceGitlabProjectHookRead(d *schema.ResourceData, meta interface{}) err
 	if err != nil {
 		return err
 	}
+
```
Comment (pdecat) [inline gitlab/resource_gitlab_project_hook.go]: Unrelated change ;)

File: gitlab/resource_gitlab_user_impersonation_token.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabUserImpersonationToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabUserImpersonationTokenCreate,
+		Read:   resourceGitlabUserImpersonationTokenRead,
+		Delete: resourceGitlabUserImpersonationTokenDelete,
+
```
Comment (pdecat) [inline gitlab/resource_gitlab_user_impersonation_token.go:23]: You should be able to also implement a pass-through importer, e.g.: https://github.com/claranet/terraform-provider-gitlab/blob/d10ba3ba63269eb0b3d9275f56e8fd7e471c5aed/gitlab/resource_gitlab_group.go#L22-L24

File: website/docs/r/user_impersonation_token.html.markdown [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+* `expires_at` - (Optional) Expiration date, format is `YYYY-MM-DD`
+
+## Attributes Reference
+
+The resource exports the following attributes:
+
+* `id` The unique id given by the Gitlab server.
+* `active` (Boolean) Is the token active or expired
+* `revoked` (Boolean) Has the token been revoked
+* `created_ad` Time of token creation
```
Comment (pdecat) [inline website/docs/r/user_impersonation_token.html.markdown]: The documentation should also describe how tokens can be imported, e.g. https://github.com/claranet/terraform-provider-gitlab/blob/claranet/website/docs/r/group.html.markdown#importing-groups

Comment (pdecat) [inline website/docs/r/user_impersonation_token.html.markdown]:
```suggestion
* `created_at` Time of token creation
```

File: gitlab/resource_gitlab_user_impersonation_token_test.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+						Revoked:   false,
+						ExpiresAt: &iso_d,
+					}),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabUserImpersonationTokenExists(n string, token *gitlab.ImpersonationToken) resource.TestCheckFunc {
```
Comment (pdecat) [inline gitlab/resource_gitlab_user_impersonation_token_test.go:107]: There should also be an import test, e.g. https://github.com/terraform-providers/terraform-provider-vault/blob/master/vault/resource_token_test.go#L49-L75

**Review by pdecat — APPROVED** [review]
LGTM!

**Comment by pdecat** [comment]
Rebased on claranet branch which was rebased on upstream master.

**Comment by pdecat** [comment]
Still WIP?

**Comment by maxenced** [comment]
> Still WIP?

nop, renamed :)
